src/Processors/IProcessor.h

```diff
@@ -189,15 +189,14 @@ class IProcessor
     /// Optimization for prepare in case we know ports were updated.
     virtual Status prepare(const PortNumbers & /*updated_input_ports*/, const PortNumbers & /*updated_output_ports*/) { return prepare(); }
 
-    /** You may call this method if 'prepare' returned Ready.
+    /** This method is a delegate to 'work' method.
+      *
+      * You may call this method if 'prepare' returned Ready.
       * This method cannot access any ports. It should use only data that was prepared by 'prepare' method.
       *
-      * Method work can be executed in parallel for different processors.
+      * Method process can be executed in parallel for different processors.
       */
-    virtual void work()
-    {
-        throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Method 'work' is not implemented for {} processor", getName());
-    }
+    virtual void process(bool trace_processors);
```

Comment:
To be honest, I believe adding method `process()` adds some confusion to users of `IProcessor` if `process()` or `work()` should be called. My preference would be to integrate the stuff in `process()` into `work()`. But maybe there are other callers of `work()` outside of `ExecutionThreadContext` and you think my proposal is not feasible.

An alternative would be to name `process()` more specific to OpenTel, e.g. `workUnderOpenTelemetrySpan()` (or whatever good name you find).